src/app/shared/redirectHelper.ts

```diff
@@ -0,0 +1,16 @@
+import { Response, Request } from "express";
+import { logError } from "./loggerHelper";
+
+export const saveSessionAndRedirect = (
+  req: Request,
+  res: Response,
+  redirectUrl: string,
+): void => {
+  req.session.save((err) => {
+    if (err) {
+      logError(req, err, "Error saving session");
+      throw err;
+    }
+    return res.redirect(redirectUrl);
+  });
+};
```

Comment:
I would suggest making this return a promise, so it's more obvious that it's actually async.

You'll need to use the Promise constructor - i.e. `return new Promise((resolve, reject) => ... );`

Reply:
This is quite right, it'll probably look something like:
```
return new Promise((resolve, reject) => {
  req.session.save(function (err) {
    if (err) {
      reject(err);
    } else {
      resolve(res.redirect(redirectUrl));
    }
  });
});
```

(Then you'll want to make sure the result is always returned and/or awaited)